Watchman.Discord/Areas/Help/Services/HelpMessageGeneratorService.cs

```diff
@@ -1,76 +1,52 @@
-            var result = this._queryBus.Execute(new GetHelpInformationQuery(contexts.Server.Id));
-
-            var lines = new Dictionary<string, string>();
-            foreach (var helpInfo in result.HelpInformations)
-            {
-                var names = helpInfo.Names.Aggregate((x, y) => $"{x} / -{y}");
-                var description = helpInfo.Descriptions.First().Details;
-
-                lines.Add(names, description);
-            }
-
-            var messageBuilder = new StringBuilder();
-            messageBuilder.PrintManyLines(lines, false);
-            return messageBuilder.ToString();
-        }
-
-        public string GenerateJsonHelp(Contexts contexts)
-        {
-            var result = this._queryBus.Execute(new GetHelpInformationQuery(contexts.Server.Id));
-
-            var serialized = JsonConvert.SerializeObject(result.HelpInformations, Formatting.Indented);
+            var serialized = JsonConvert.SerializeObject(helpInformations, Formatting.Indented);
             serialized = RemoveFirstAndLastBracket(serialized);
             return serialized;
         }
 
         public IEnumerable<KeyValuePair<string, string>> MapToEmbedInput(IEnumerable<HelpInformation> helpInformations)
         {
-            return helpInformations.Select(x =>
+            foreach(var helpInfo in helpInformations)
             {
-                var name = "-" + x.Names.First();
+                var name = "-" + helpInfo.Names.Aggregate((a, b) => $"{a} / {b}");
+
+                var descriptions = helpInfo.Descriptions
+                    .Where(x => x.Details.Trim().ToLowerInvariant() != "empty")
+                    .Select(x => $"{x.Name} => {x.Details}")
+                    .ToList();
 
-                var descriptions = x.Descriptions.Where(x => x.Details.Trim().ToLowerInvariant() != "empty").Select(x => $"{x.Name} => {x.Details}");
-                var arguments = x.ArgumentInfos.Where(x => x.Description.Trim().ToLowerInvariant() != "empty")?.Select(x => $"{x.Name} => {x.Description}");
+                var arguments = helpInfo.ArgumentInfos
+                    .Where(x => x.Description.Trim().ToLowerInvariant() != "empty")
+                    .Select(x => $"{x.Name} => {x.Description}")
+                    .ToList();
 
-                var content = descriptions.Any() ? descriptions.Aggregate((a, b) => a + "\n" + b) : string.Empty;
-                if(!string.IsNullOrWhiteSpace(content))
+                var content = "";
+                if(descriptions.Any())
                 {
-                    content = content + "\n\n";
+                    var commandDescription = descriptions.Aggregate((a, b) => a + "\n" + b);
+                    content = commandDescription + "\n\n";
                 }
                 if(arguments.Any())
                 {
-                    content = content + arguments.Aggregate((a, b) => a + "\n" + b);
+                    content += arguments.Aggregate((a, b) => a + "\n" + b);
```

Comment:
+= dodaje na początek a nie na koniec z tego co pamiętam

Reply:
Mi się wydaje, że to jest tożsame, a poza tym nawet sam resharper mi to zaproponował, więc zakładam że jest świadom tego jak działa